src/main/java/org/elasticsearch/common/joda/Joda.java

```diff
@@ -133,6 +134,10 @@ public static FormatDateTimeFormatter forPattern(String input, Locale locale) {
             formatter = ISODateTimeFormat.yearMonth();
         } else if ("yearMonthDay".equals(input) || "year_month_day".equals(input)) {
             formatter = ISODateTimeFormat.yearMonthDay();
+        } else if ("epoch_second".equals(input)) {
```

Comment:
I just realized we aren't even talking about setting names, but the valid values for the `format` setting. This argument to use ParseValue does not make sense. We don't support camelCase in eg the `index` option. We should not do it here, it will just add more work for users if we allow them to *start* using a new value that will just go away in the future (and will require them to change the value to what they would have found in the first place if they had tried using camelCase and seen an error).